app/script/components/list/conversationListCallingCell.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.ConversationListCallingCell = class ConversationListCallingCell {
+  constructor(params) {
+    this.conversation = params.conversation;
+    this.calling_repository = params.calling_repository;
+    this.is_selected = params.is_selected;
+
+    this.on_accept_call = () => {
+      amplify.publish(z.event.WebApp.CALL.STATE.JOIN, this.conversation.id, false);
+    };
+
+    this.on_accept_video = () => {
+      amplify.publish(z.event.WebApp.CALL.STATE.JOIN, this.conversation.id, true);
+    };
+
+    this.on_leave_call = () => {
+      amplify.publish(z.event.WebApp.CALL.STATE.LEAVE, this.conversation.id, z.calling.enum.TERMINATION_REASON.SELF_USER);
+    };
+
+    this.on_reject_call = () => {
+      amplify.publish(z.event.WebApp.CALL.STATE.REJECT, this.conversation.id);
+    };
+
+    this.on_toggle_audio = () => {
+      amplify.publish(z.event.WebApp.CALL.MEDIA.TOGGLE, this.conversation.id, z.media.MediaType.AUDIO);
+    };
+
+    this.on_toggle_screen = () => {
+      amplify.publish(z.event.WebApp.CALL.MEDIA.CHOOSE_SCREEN, this.conversation.id);
+    };
+
+    this.on_toggle_video = () => {
+      amplify.publish(z.event.WebApp.CALL.MEDIA.TOGGLE, this.conversation.id, z.media.MediaType.VIDEO);
+    };
+
+    this.users = ko.pureComputed(() => this.conversation.participating_user_ets());
+    this.call = ko.pureComputed(() => this.conversation.call());
+    this.call_participants = ko.pureComputed(() => {
+      return this.call().participants()
+        .map((participant_et) => participant_et.user)
+        .filter((user_et) => !user_et.is_me);
```

Comment:
Is this necessary? The self user should never be among the call participants. If you saw that, it is a bug in my code.